src/main/java/seedu/address/model/person/medicalrecord/Note.java

```diff
@@ -1,50 +1,54 @@
 package seedu.address.model.person.medicalrecord;
 
-import static java.util.Objects.requireNonNull;
-import static seedu.address.commons.util.AppUtil.checkArgument;
+import java.util.Map;
+import java.util.Objects;
+
+import seedu.address.model.medicine.SerialNumber;
 
 /**
  * Represents a note object belonging to a patient.
  */
 public class Note {
 
-    public static final String MESSAGE_NOTE_CONSTRAINTS =
-            "Notes should not be blank, and first character should not be a whitespace.";
-    public static final String NOTE_VALIDATION_REGEX = "[^\\s].*";
-    public final String value;
-
-    /**
-     * Constructs a {@code Note}.
-     *
-     * @param note A valid note.
-     */
-    public Note(String note) {
-        requireNonNull(note);
-        checkArgument(isValidNote(note), MESSAGE_NOTE_CONSTRAINTS);
-        value = note;
+    // Items belonging to a note object
+    private Message message;
+    private Map<SerialNumber, Quantity> dispensedMedicines;
+
+    public Note(Message message, Map<SerialNumber, Quantity> dispensedMedicines) {
+        this.message = message;
+        this.dispensedMedicines = dispensedMedicines;
     }
 
-    /**
-     * Returns true if a given string is a valid note.
-     */
-    public static boolean isValidNote(String test) {
-        return test.matches(NOTE_VALIDATION_REGEX);
+    public Message getMessage() {
+        return message;
     }
 
-    @Override
-    public String toString() {
-        return value;
+    public void setMessage(Message message) {
+        this.message = message;
+    }
+
+    public Map<SerialNumber, Quantity> getDispensedMedicines() {
+        return dispensedMedicines;
+    }
+
+    public void setDispensedMedicines(Map<SerialNumber, Quantity> dispensedMedicines) {
+        this.dispensedMedicines = dispensedMedicines;
     }
 
     @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Note // instanceof handles nulls
-                && value.equals(((Note) other).value)); // state check
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("Note: ")
+                .append(" Message: ")
+                .append(getMessage())
+                .append(" Dispensed Medicines: ")
```

Comment:
Think you should add a \n to the start here? For ease of readability to divide message and the dispensed medicines.